packages/react-sdk/src/core/contexts/MediaDevicesContext.tsx

```diff
@@ -294,10 +311,10 @@ export const MediaDevicesProvider = ({
     return () => subscription.unsubscribe();
   }, [videoDevices.length]);
 
-  // useEffect(() => {
-  //   if (!call) return;
-  //   call.setAudioOutputDevice(selectedAudioOutputDeviceId);
-  // }, [call, selectedAudioOutputDeviceId]);
+  useEffect(() => {
+    if (!call || callingState !== CallingState.JOINED) return;
+    call.setAudioOutputDevice(selectedAudioOutputDeviceId);
```

Comment:
I think this may break lobby?

Reply:
I don't think it does. The device selector there just stores the user's preference. The `VideoPreview` doesn't emit any audio.